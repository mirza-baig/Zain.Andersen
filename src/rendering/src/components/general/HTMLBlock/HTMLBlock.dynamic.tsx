// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
// Components
import { Component } from 'src/helpers/Component';
import { ComponentProps } from 'lib/types/component-props';
import { useEffect, useRef } from 'react';

export type HTMLBlockProps =
  Feature.EnterpriseWeb.Enterprise.Components.General.HtmlBlock.HtmlBlock & ComponentProps;

const HTMLBlock = (props: HTMLBlockProps) => {
  const divRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    if (props.fields?.code?.value) {
      try {
        const fragment = document.createRange().createContextualFragment(props.fields?.code?.value);
        divRef.current?.append(fragment);
      } catch (error) {
        console.error('Error creating fragment:', error);
      }
    }
    // Suggested deps are coming from layout service. We can ignore useEffect warning.
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  if (!props.fields?.code?.value) {
    return <></>;
  }

  return (
    <Component variant="full" dataComponent="general/htmlblock" {...props}>
      <div className="col-span-12" ref={divRef}></div>
    </Component>
  );
};

export default withDatasourceCheck()<HTMLBlockProps>(HTMLBlock);
