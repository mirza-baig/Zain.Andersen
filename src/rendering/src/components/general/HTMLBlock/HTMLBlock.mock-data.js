const defaultData = {
  rendering: {
    componentName: 'HTMLBlock',
    dataSource:
      '/sitecore/content/AndersenCorporation/RenewalByAndersen/RenewalByAndersen/Home/Components/HTML Block/Data/Curalate Gallery',
  },
  fields: {
    code: {
      value:
        '<script>\r\nvar CRL8_SITENAME = \'renewalbyandersen-sb1oge\'; !function(){var e=window.crl8=window.crl8||{},n=!1,i=[];e.ready=function(e){n?e():i.push(e)},e.pixel=e.pixel||function(){e.pixel.q.push(arguments)},e.pixel.q=e.pixel.q||[];var t=window.document,o=t.createElement("script"),c=e.debug||-1!==t.location.search.indexOf("crl8-debug=true")?"js":"min.js";o.async=!0,o.src=t.location.protocol+"//edge.curalate.com/sites/"+CRL8_SITENAME+"/site/latest/site."+c,o.onload=function(){n=!0,i.forEach(function(e){e()})};var r=t.getElementsByTagName("script")[0];r.parentNode.insertBefore(o,r.nextSibling)}();\r\n</script>\r\n<div data-crl8-container-id="product" data-crl8-filter="productId:\'awning\'"></div>',
    },
  },
};

export default defaultData;
