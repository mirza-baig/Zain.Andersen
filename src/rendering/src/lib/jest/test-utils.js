// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
