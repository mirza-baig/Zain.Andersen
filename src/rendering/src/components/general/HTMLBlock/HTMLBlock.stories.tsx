// Global
import { Story, Meta } from '@storybook/react';
// Local
import HTMLBlock, { HTMLBlockProps } from './HTMLBlock.dynamic';
import defaultData from './HTMLBlock.mock-data';

export default {
  title: 'General/HTMLBlock',
  component: HTMLBlock,
} as Meta;
const Template: Story<HTMLBlockProps> = (props) => <HTMLBlock {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
