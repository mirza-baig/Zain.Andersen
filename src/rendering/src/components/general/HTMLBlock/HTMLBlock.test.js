// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import HTMLBlock from './HTMLBlock.dynamic';
import defaultData from './HTMLBlock.mock-data';

it('renders correctly', async () => {
  const component = snapshot(HTMLBlock, { componentProps: defaultData });
  await hasDataComponent(component, 'general/htmlblock');
});
