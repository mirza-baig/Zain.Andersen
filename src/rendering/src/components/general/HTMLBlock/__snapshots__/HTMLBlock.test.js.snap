// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/htmlblock"
  id="id2077084623"
>
  <div
    class="my-8 w-full theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-m gap-s md:gap-s "
    >
      <div
        class="col-span-12"
      >
        <script>
          
var CRL8_SITENAME = 'renewalbyandersen-sb1oge'; !function(){var e=window.crl8=window.crl8||{},n=!1,i=[];e.ready=function(e){n?e():i.push(e)},e.pixel=e.pixel||function(){e.pixel.q.push(arguments)},e.pixel.q=e.pixel.q||[];var t=window.document,o=t.createElement("script"),c=e.debug||-1!==t.location.search.indexOf("crl8-debug=true")?"js":"min.js";o.async=!0,o.src=t.location.protocol+"//edge.curalate.com/sites/"+CRL8_SITENAME+"/site/latest/site."+c,o.onload=function(){n=!0,i.forEach(function(e){e()})};var r=t.getElementsByTagName("script")[0];r.parentNode.insertBefore(o,r.nextSibling)}();

        </script>
        

        <div
          data-crl8-container-id="product"
          data-crl8-filter="productId:'awning'"
        />
      </div>
    </div>
  </div>
</section>
`;
